packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -649,6 +649,22 @@ export class GarbageCollector implements IGarbageCollector {
          */
         this.initializeBaseStateP = new LazyPromise<void>(async () => {
             const currentReferenceTimestampMs = this.runtime.getCurrentReferenceTimestampMs();
+            /**
+             * If there is no current reference timestamp, skip initialization. We need the current timestamp to track
+             * how long objects have been unreferenced and if they can be deleted.
+             *
+             * Note that the only scenario where there is no reference timestamp is when no ops have ever been processed
+             * for this container and it is in read mode. In this scenario, there is no point in running GC anyway
+             * because references in the container do not change without any ops, i.e., there is nothing to collect.
+             */
+            if (currentReferenceTimestampMs === undefined) {
+                // Log an event so we can evaluate how often we run into this scenario.
+                this.mc.logger.sendErrorEvent({
+                    eventName: "GarbageCollectorInitializedWithoutTimestamp",
+                });
+                return;
+            }
+
             const baseState = await baseSummaryStateP;
             if (baseState === undefined) {
```

Comment:
How does this condition relate to the "no timestamp" condition?  Worth adding a comment or log here?

Reply:
They are not related. This means that there was no GC state in the summary. This can happen when either old documents are loaded before GC was a thing or a document that has GC disabled is loaded.

Reply:
I will add a comment.